packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -4248,3 +4248,28 @@ class RenderFollowerLayer extends RenderProxyBox {
     properties.add(new TransformProperty('current transform matrix', getCurrentTransform()));
   }
 }
+
+/// Render object which inserts an [AnnotatedRegionLayer] into the layer tree.
+///
+/// See also:
+///
+///   * [Layer.find], for an example of how this value is retrieved.
+///   * [AnnotatedRegionLayer], the layer this render object creates.
+class AnnotatedRegionRenderObject<T> extends RenderProxyBox {
+  /// A value which can be retrieved using [Layer.find].
+  T value;
+
+  /// Whether the render object will pass its [size] to the [AnnotatedRegionLayer].
+  bool sized;
+
+  @override
+  final bool alwaysNeedsCompositing = true;
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child != null) {
+      final AnnotatedRegionLayer<T> layer = new AnnotatedRegionLayer<T>(value, size: sized ? size : null);
+      context.pushLayer(layer, super.paint, offset);
+    }
```

Comment:
why do you only push the layer if there's a child?

Reply:
Fixed